crates/core/src/execution.rs

```diff
@@ -1,20 +1,40 @@
 use std::process;
 
-use anyhow::{bail, Error, Result};
+use anyhow::{anyhow, bail, Error, Result};
 use javy::{
     from_js_error,
     quickjs::{context::EvalOptions, Module, Value},
-    Runtime,
+    to_js_error, Runtime,
 };
 
+static EVENT_LOOP_ERR: &str = r#"
+                Pending jobs in the event queue.
+                Scheduling events is not supported when the 
+                experimental_event_loop cargo feature is disabled.
+            "#;
+
 /// Evaluate the given bytecode.
 ///
 /// Evaluating also prepares (or "instantiates") the state of the JavaScript
 /// engine given all the information encoded in the bytecode.
 pub fn run_bytecode(runtime: &Runtime, bytecode: &[u8]) {
     runtime
         .context()
-        .with(|this| Module::instantiate_read_object(this, bytecode).map(|_| ()))
+        .with(|this| {
+            let module = unsafe { Module::load(this.clone(), bytecode)? };
+            let (_, promise) = module.eval()?;
+
+            if cfg!(feature = "experimental_event_loop") {
+                // If the experimental event loop is enabled, trigger it.
+                promise.finish::<Value>().map(|_| ())
```

Comment:
What does the `.map(|_| ())` do? Could we just do `promise.finish::<Value>();` if we want to swallow the return value?

Reply:
That will introduce incompatible return types for if/else: `()` vs `Result<()>`